subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -51,7 +51,7 @@
 public class InitBuild extends DefaultTask {
     private final Directory projectDir = getProject().getLayout().getProjectDirectory();
     private String type;
-    private final Property<Boolean> splitProject = getProject().getObjects().property(Boolean.class);
```

Comment:
I don't think we should change this to a String.  Could we just give this a convention of false ?

Another idea would be to make the way we parse `@Option`s to support a "no" syntax like we have with other command-line options.

e.g., you could do `init --type java-application --split-project ...` to generate a split project and `init --type java-application --no-split-project ...` to generate a non-split project.

This is something that's been asked for before.

Reply:
I agree that we should not change this to a String. And that we should probably have something like a `--no...` option if we want to solve it this way.

However, for this case I would solve it differently. I see `--split-project` as a detail for `-type` and not a completely independent option. So I would go for the following behavior:

1. `--type java-application --split-project` - Do _application_ with split
2. `--type java-application` - Do _application_ **without** _split_ (do **not** ask)
3. (no `--type` defined) - Ask

So I would keep the original code and add a condition:

```
else if (!isNullOrEmpty(type)) {
   modularizationOption = false;
} else {
   ...
} 
```